### PR TITLE
SONAR-8894 Search members of an organization WS api/organizations/search_members

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/organization/ws/SearchMembersAction.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+  private static SearchOptions buildSearchOptions(Request request) {
+    int pageSize = request.mandatoryParamAsInt(Param.PAGE_SIZE);
+    checkArgument(pageSize <= SearchOptions.MAX_LIMIT, "Page size must lower than %s", SearchOptions.MAX_LIMIT);
+
+    return new SearchOptions().setPage(request.mandatoryParamAsInt(Param.PAGE), pageSize);
+  }
+
+  private static String hash(String text) {
+    return Hashing.md5().hashString(text.toLowerCase(Locale.ENGLISH), StandardCharsets.UTF_8).toString();
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/organization/ws/SearchMembersAction.java]: CRITICAL Use a stronger hashing algorithm than MD5. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2070)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'

File: server/sonar-server/src/main/java/org/sonar/server/organization/ws/SearchMembersAction.java [hunk 2]
```diff
@@ -20,13 +20,56 @@
...
+  public SearchMembersAction(DbClient dbClient, UserIndex userIndex, DefaultOrganizationProvider organizationProvider, UserSession userSession) {
+    this.dbClient = dbClient;
+    this.userIndex = userIndex;
+    this.organizationProvider = organizationProvider;
+    this.userSession = userSession;
+  }
+
   @Override
   public void define(WebService.NewController context) {
     WebService.NewAction action = context.createAction("search_members")
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/organization/ws/SearchMembersAction.java]: Description is missing

File: server/sonar-server/src/main/java/org/sonar/server/organization/ws/SearchMembersAction.java [hunk 3]
```diff
@@ -47,6 +95,90 @@ public void define(WebService.NewController context) {
...
+      .setTotal((int) searchResults.getTotal())
+      .build();
+  }
+
+  private OrganizationDto getOrganization(DbSession dbSession, @Nullable String organizationParam) {
+    String organizationKey = Optional.ofNullable(organizationParam)
+      .orElseGet(organizationProvider.get()::getKey);
+    return checkFoundWithOptional(
+      dbClient.organizationDao().selectByKey(dbSession, organizationKey),
+      "No organizationDto with key '%s'", organizationKey);
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/organization/ws/SearchMembersAction.java]: Type : "organizationDto" -> "organization"

**Comment by julienlancelot** [comment]
No checks are done on the size of the query parameter

**Pull request closed without merging**